template-share-extension/src/extension/Share.tsx

```diff
@@ -0,0 +1,145 @@
+                            size={Size.m}
+                        />
+
+                        <div style={{ margin: '8px 0' }}>
+                            <TextField
+                                multiline
+                                value={message}
+                                onChange={setMessage}
+                                label="Your message"
+                                maxLength={5000}
+                            />
+                        </div>
+
+                        <LinkPreview
+                            title={title}
+                            description="Description"
+                            link={link}
+                            theme={theme}
+                            thumbnailProps={{
+                                image:
+                                    'https://images.unsplash.com/photo-1549492423-400259a2e574?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=554&q=80',
+                                alt: 'Landscape',
+                            }}
+                        />
+                    </div>
+                </FlexBox>
+
+                <Toolbar
+                    style={{ paddingRight: 0 }}
+                    after={
+                        <FlexBox orientation={Orientation.horizontal}>
+                            <Button
+                                onClick={onClose}
+                                emphasis={Emphasis.medium}
+                                className="lumx-spacing-margin-right-regular"
+                            >
+                                Cancel
+                            </Button>
+                            <Button
+                                onClick={() => {
+                                    alert(`Share content ${uid}`);
+                                }}
+                                emphasis={Emphasis.high}
+                                disabled={!message || message.length === 0}
+                            >
+                                Share
+                            </Button>
+                        </FlexBox>
+                    }
+                />
+            </FlexBox>
+        </div>
+    );
+};
+
+const NotificationAwareContent = (props: any) => {
+    const messages: any = {
+        en: messagesEn,
+        fr: messagesFr,
+    };
+    const [lang, setLang] = useState<string>('en');
+    useEffect(() => {
+        const getContext = async () => {
+            const lumapps = new Lumapps();
+            const { userLang: userLangPromise } = lumapps.context;
+
+            const userLang = await userLangPromise;
+            const isLangInTrad = Object.keys(messages).includes(userLang);
+
+            setLang(isLangInTrad ? userLang : 'en');
+        };
+        getContext();
+    }, [messages]);
+
+    return (
+        <IntlProvider messages={messages[lang]} locale={lang}>
```

Comment:
You can then update code like this:
`<IntlProvider messages={messages[lang as keyof typeof messages]} locale={lang}>`